src/components/editor/metaform-editor.tsx

```diff
@@ -37,14 +38,13 @@ const MetaformEditor: React.FC<Props> = ({
   const [ selectedFieldIndex, setSelectedFieldIndex ] = React.useState<number>();
   const [ selectedSectionIndex, setSelectedSectionIndex ] = React.useState<number>();
   const [ draggingMode, setDraggingMode ] = React.useState<DraggingMode>();
-
+  
   /**
    * Event handler for empty space click
    */
   const onGlobalClick = (event: MouseEvent) => {
     if (!editorRef.current?.contains(event.target as Node)) {
-      setSelectedFieldIndex(undefined);
-      setSelectedSectionIndex(undefined);
+      return null;
```

Comment:
If globalclick set field- and selectionIndex undefined, we lose our right drawer field data.
Because when we click right drawer textfield to change title it thinks its globalclick and set indexes undefined which updates right drawer to show nothing.